python/tskit/tables.py

```diff
@@ -46,51 +45,49 @@
 dataclass_options = {"frozen": True}
 
 
-@dataclass(eq=False, **dataclass_options)
+@metadata.lazy_decode
+@dataclass(**dataclass_options)
 class IndividualTableRow:
     __slots__ = ["flags", "location", "parents", "metadata"]
     flags: int
     location: np.ndarray
     parents: np.ndarray
-    metadata: bytes
+    metadata: Any
```

Comment:
We could say ``metadata : Union[bytes, dict]`` couldn't we?

Reply:
Changed to `Optional[Union[bytes, dict]]`